src/core/stream_send.c

```diff
@@ -249,6 +249,34 @@ QuicStreamSendShutdown(
         }
         Stream->Flags.LocalCloseResetReliable = TRUE;
         Stream->SendShutdownErrorCode = ErrorCode;
+
+        while (ApiSendRequests != NULL) {
+            //
+            // App queued some data around the same time it called shutdown. Bad App!
+            // Enqueue all the stuff in ApiSendRequests to SendRequests.
+            //
+            QUIC_SEND_REQUEST* SendRequest = ApiSendRequests;
+            ApiSendRequests = ApiSendRequests->Next;
+            SendRequest->Next = NULL;
+            QuicStreamEnqueueSendRequest(Stream, SendRequest);
+
+            //
+            // Update flags.
+            //
+            QuicSendSetStreamSendFlag(
+                &Stream->Connection->Send,
+                Stream,
+                QUIC_STREAM_SEND_FLAG_DATA,
+                !!(SendRequest->Flags & QUIC_SEND_FLAG_DELAY_SEND));
```

Comment:
I don't think we should need to set the flags in this case, because we know we don't need to send it (we're aborting).

Reply:
Removed! 2d8415f6bf02af962e461fc41b8ef9b23b9ab669